dask_sql/physical/rel/logical/limit.py

```diff
@@ -101,8 +88,8 @@ def select_from_to(df, partition_index, partition_borders):
 
             return df.iloc[from_index:to_index]
 
-        # (b) Now we just need to apply the function on every partition
-        # We do this via the delayed interface, which seems the easiest one.
-        return map_on_partition_index(
-            df, select_from_to, partition_borders, meta=df._meta
+        return df.map_partitions(
+            limit_partition_func,
+            BlockIndex(numblocks=(df.npartitions,)),
```

Comment:
iiuc, the `partition_index` attribute associates a partition with a number/value. Is there a reason why `BlockIndex` is preferred in this situation?

Reply:
> iiuc, the partition_index attribute associates a partition with a number/value

Correct, `partition_index` is expected to be the index of the partition within the overall Dask dataframe - the first partition should have `partition_index=0`, the second `partition_index=1`, and so on.

> Is there a reason why `BlockIndex` is preferred in this situation?

My understanding here is the "blocks" of a Dask dataframe are its partitions, making `BlockIndex` roughly synonymous with partition index here.

Chatting with @rjzamora, I now know that another approach to getting partition index would be to add kwarg `partition_info` to the function, which would then be populated with the partition index, among some other info. Perhaps that approach might be a little clearer to future developers?

Reply:
I don't have a strong preference here and `BlockIndex` getting the same info from dask should be good for the sake of this pr.
For me personally, explicitly mentioning the argument name when calling `map_partitions` should be enough to give an idea of what this is, something like `df.map_partitions(..., partition_index=BlockIndex(numblocks=(df.npartitions,))`.

More generally, the reason for my confusion was primarily because I've seen `Block` like terminology used in the context of IO in dask and `partition` like terminology used in the context of dataframes.

Reply:
> something like `df.map_partitions(..., partition_index=BlockIndex(numblocks=(df.npartitions,))`.

Unfortunately, `map_partitions` has different handling for args versus kwargs that makes it so we must supply the `BlockIndex` as an argument in order for it to be computed for the partition functions.

I'll try out `partition_info`, since it makes it somewhat clearer what is being done and seems to be the more documented way of achieving what we want (getting partition info within the function).

Reply:
Using `partition_info` is fine.  Under the hood, it will just add on a `BlockwiseDep` argument (one that is a bit heavier weight than `BlockIndex`).  I *think* you should feel comfortable going that route here.

If it turns out that you need to use `BlockIndex` directly in the future, you can probably make the code a bit more intuitive by naming the argument a bit more clearly:

```python
partition_indices = BlockIndex(numblocks=(df.npartitions,)
df.map_partitions(..., partition_indices)
```